examples/hello-world/step-by-step/cifar10/sag_with_deploy_map/sag_deploy_map.ipynb

```diff
@@ -0,0 +1,282 @@
+    "  * [Convert training code to federated learning training code](../sag/sag.ipynb#code)\n",
+    "  * [Prepare Data](../sag/sag.ipynb#data)\n",
+    "\n",
+    "\n",
+    "The next step is to preapre job folder and configurations. We will start from here. \n",
+    "\n",
+    "## Job Folder and Configurations\n",
+    "\n",
+    "Before we dive into the details of the job configurations, let's first introduce few terminologies.\n",
+    "\n",
+    "### Job and App\n",
+    "\n",
+    "So far, when we referring NVFLARE job, we are referring the job folder which consists \n",
+    "\n",
+    "  * The job folder structure\n",
+    "  * The job configuration and \n",
+    "  * The custom codes \n",
+    "\n",
+    "Actually, the complete picture of the job is the followings:\n",
+    "\n",
+    "  * The job has special folder structure\n",
+    "  * The job is consists of many \"apps\" or \"applications\"\n",
+    "  * each application is consists app's custom code and configurations. \n",
+    "  \n",
+    "  you probably has noticed the folder structure as \n",
+    "```\n",
+    "  <job folder>\n",
+    "  | -- app\n",
+    "  | -----config/\n",
+    "  | --------config_fed_client.conf\n",
+    "  | --------config_fed_server.conf\n",
+    "  | -----custom\n",
+    "  | --------train.py\n",
+    "    \n",
+    "```\n",
+    "  In such case, we only defined one app. We could define many apps. \n",
+    "  \n",
+    "  ```\n",
+    "  <job folder>\n",
+    "  | -- app_server\n",
+    "  | -- app_1\n",
+    "  | -- app_2\n",
+    "    \n",
+    "```\n",
+    "  in above,we defined three apps, where server, site-1, site-2 each has different app. \n",
+    "   \n",
+    "### deploy_map\n",
+    "\n",
+    "What's \"deploy_map\" ? It is the \"application to site deployment mapping\". The deploy_map defines where to deploy these apps. \n",
+    "\n",
+    "For examaple, in meta.conf (or meta.json), you probably seen something like this: \n",
+    "\n",
+    "```\n",
+    "{\n",
+    "    \"name\": \"my_job\",\n",
+    "    \"resource_spec\": {},\n",
+    "    \"deploy_map\": {\n",
+    "        \"app\": [\n",
+    "            \"@ALL\"\n",
+    "        ]\n",
+    "    },\n",
+    "    \"min_clients\": 1,\n",
+    "    \"mandatory_clients\": [],\n",
+    "}\n",
+    "\n",
+    "\n",
+    "```\n",
+    "The deploy_map here\n",
+    "```\n",
+    "    \"deploy_map\": {\n",
+    "        \"app\": [\n",
+    "            \"@ALL\"\n",
+    "        ]\n",
+    "```\n",
+    "specify that the application \"app\" will be deployed to ALL sites. \"@ALL\" is special symbol, short-hand for all participanting clients and server. \n",
+    "\n",
+    "\n",
+    "Here is another example of deploy_map\n",
+    "\n",
+    "There are three \"app\"s, each named \"app1\", \"app2\", and \"app3\".  The deploy_map rules states, that \"app1\" is only deploy to \"server\" site, \"app2\" and \"app3\" will be deployed to site-1 and site-2 respectively. \n",
```

Comment:
"There are three \"app\"s: \"app_server\", \"app_1\", and \"app_2\". The deploy_map rules states, that \"app_server\" is only deploy to \"server\" site, \"app_1\" and \"app_2\" will be deployed to site-1 and site-2 respectively. \n"